### PR TITLE
(Mostly) add AliasEq and start porting chalk projection tests

### PR DESCRIPTION
**Pull request by jackh726**
A decent start :)

### CODE REVIEW

File: tests/projection.rs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        }
+    }
+]";
+
+
+#[test]
+fn projection_equality() {
+    expect_test::expect![[r#"
+        Ok(
+            {},
```
Comment (jackh726) [inline tests/projection.rs]: This isn't right :) Should be ?0 = u32. The first goal (the AliasEq) finds this constraint, but after solving the `S: Trait1<>` goal we don't have any

File: tests/projection.rs [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        )
+    "#]]
+    .assert_debug_eq(&test_where_clause(
+        PROJECTION_EQUALITY,
+        "exists<ty U> {} => { S: Trait1<>, <S as Trait1<>>::Type<> = U }",
```
Comment (jackh726) [inline tests/projection.rs]: Removing the `S: Trait1<>` gives the expected constraints!

File: tests/projection.rs (hunk 1 above)
Comment (compiler-errors) [reply tests/projection.rs]: This is an outlives predicate

File: crates/formality-prove/src/prove/prove_wf.rs [hunk 3]
```diff
@@ -39,10 +39,33 @@ judgment_fn! {
             (prove_wf(decls, env, assumptions, RigidTy { name: RigidName::ScalarId(_), parameters }) => c)
         )
 
+        (
+            (for_all(&decls, &env, &assumptions, &parameters, &prove_wf) => c)
+            --- ("integers and booleans")
```
Comment (nikomatsakis) [inline crates/formality-prove/src/prove/prove_wf.rs]:
```suggestion
            --- ("ADT")
```

Comment (nikomatsakis) [reply crates/formality-prove/src/prove/prove_wf.rs]: we should be checking the where-clauses on the ADT declaration, but let's leave that for a follow-up PR

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**